### DISCUSSION TITLE
Install as a editable package failed: [KeyError]: "There is no item named '<name>-<version>.editable.dist-info/RECORD' in the archive"

### DISCUSSION THREAD

**Discussion by ghost** [pdm-project/pdm#2208] [Q&A] [answered]
Sometimes if I do `pdm install` to install self as editable package, KeyError occurred.
```
Installing the project as an editable package...
/home/runner/.local/share/pdm/venv/lib/python3.10/site-packages/pdm/installers/installers.py:174: UserWarning: Validation of the RECORD file of /tmp/pdm-wheel-cache-8_kn1vqf/04/42/2f/430dbd42ee398ac0416ebdbc67ca401fc1840f1b592cc98c3c/pyfsd-0.0.1.dev7+gdc4b726.d20230828.editable-py3-none-any.whl failed. Please report to the maintainers of that package so they can fix their build process. Details:
Unable to retrieve `RECORD` from /tmp/pdm-wheel-cache-8_kn1vqf/04/42/2f/430dbd42ee398ac0416ebdbc67ca401fc1840f1b592cc98c3c/pyfsd-0.0.1.dev7+gdc4b726.d20230828.editable-py3-none-any.whl: KeyError("There is no item named 'pyfsd-0.0.1.dev7+gdc4b726.editable.dist-info/RECORD' in the archive")

  _install_wheel(wheel=wheel, destination=destination, additional_metadata=additional_metadata)
  ✖ Install pyfsd 0.0.1.dev7+gdc4b726.editable failed

See /tmp/pdm-install-ua7_u_c1.log for detailed debug log.
[KeyError]: "There is no item named 'pyfsd-0.0.1.dev7+gdc4b726.editable.dist-info/RECORD' in the archive"
```
[pyproject.toml](https://github.com/cfcsim/pyfsd/blob/dc4b726feb5234f5b4d83e4c12cfd7d19b0edc9e/pyproject.toml)  
Have I did anything wrong?  

--------
[pdm-install-ua7_u_c1.log](https://github.com/cfcsim/pyfsd/suites/15561111324/artifacts/887980698)
[reproduce steps (github action)](https://github.com/cfcsim/pyfsd/blob/dc4b726feb5234f5b4d83e4c12cfd7d19b0edc9e/.github/workflows/python.yml)

--------
*python3.8 seem ok

**Comment by pvcnt** [comment]
I am having the same error, it fails on my CI but not my laptop, but I still have no clue why. A similar error was reported here, but issue was closed: https://github.com/pdm-project/pdm/issues/2272

**Comment by pawamoy** [comment]
Happens to me too sometimes I think. I generally remove and reinstall all deps to fix it.

**Reply by ghost** [reply]
I guess: when I run `pdm install`, pdm generate version from scm, then it write version into src/pyfsd/_version.py. After that, pdm generate version again to install editable package, then version generator found that src/pyfsd/_version.py changed so append `.d2023xxxx` to the version, then error occur

**Reply by frostming** [reply]
> I guess: when I run `pdm install`, pdm generate version from scm, then it write version into src/pyfsd/_version.py. After that, pdm generate version again to install editable package, then version generator found that src/pyfsd/_version.py changed so append `.d2023xxxx` to the version, then error occur

no, pdm-backend writes version file to a temporary .pdm-build folder, which can be checked after build done. So I need to know the exact steps and files to reproduce

* * *

Oh I know what is happening. Please remember to gitignore the generated file since editable build writes version file in place!

**Reply by ghost** [reply]
Yes, if I run `pdm install`, `src/pyfsd/_version.py` is changed. But how can I stop pdm from doing that?

**Reply by frostming** [reply]
Hmm, I still can't reproduce, even with the file updated in-place, the version number doesn't change.

**Reply by ghost** [reply]
[Reproduce](https://github.com/gamecss/reproduce1)
PDM 2.9.3 with cpython 3.10.12, just clone it and execute `pdm install`

**Reply by pawamoy** [reply]
Prefer "gentle ping" instead of "hello?" :wink: Maintainers do their best to help everybody, it's easy to forget about an ongoing discussion.

**Reply by ghost** [reply]
I'm so so sorry...... My English is poor

**Reply by pawamoy** [reply]
No worries!

**Reply by frostming** [reply]
The reproduce repo is really useful in letting me know what I am blind of. A fix is on the way in `pdm-backend`

**Reply by frostming** [reply, marked as answer]
`pdm-backend` 2.1.7 is released, it can be automatically pulled when you run the build.
